library/dokku_image.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.dokku_app import dokku_app_ensure_present
+
+DOCUMENTATION = """
+---
+module: dokku_image
+short_description: Pull Docker image and deploy app
+options:
+  app:
+    description:
+      - The name of the app
+    required: True
+    default: null
+    aliases: []
+  image:
+    description:
+      - Docker image
+    required: True
+    default: null
+    aliases: []
+  user_name:
+    description:
+      - Git user.name for customizing the author's name
+    required: False
+    default: null
+    aliases: []
+  user_email:
+    description:
+      - Git user.email for customizing the author's email
+    required: False
+    default: null
+    aliases: []
+  build_dir:
+    description:
+      - Specifiy custom build directory for a custom build context
+    required: False
+    default: null
+    aliases: []
+author: Simo Aleksandrov
+"""
+
+EXAMPLES = """
+- name: Pull and deploy meilisearch
+  dokku_image:
+      app: meilisearch
+      image: getmeili/meilisearch:v0.24.0rc1
+- name: Pull and deploy image with custom author
+  dokku_image:
+      app: hello-world
+      user_name: Elliot Alderson
+      user_email: elliotalderson@protonmail.ch
+      repository: hello-world:latest
+- name: Pull and deploy image with custom build dir
+  dokku_image:
+      app: hello-world
+      build_dir: /path/to/build
+      repository: hello-world:latest
+"""
+
+
+def dokku_git_sha(data):
```

Comment:
If I see correctly, you're copy-pasting this function. If that's the case, I suggest you instead move it to a new `module_utils/dokku_git.py` file and import it here and in `dokku_clone.py`

Reply:
By the way, I notice that `dokku_git_sha` actually only uses `data["app"]`.

As a bonus, you could make its argument `app`